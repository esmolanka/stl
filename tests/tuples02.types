
#check (Int, String) <: (Int,)

#check (Int, String, Float) <: exists a. (Int, a)
