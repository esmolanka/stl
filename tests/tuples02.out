tests/tuples02.types:2:1-2:29:
  Subtype:
    ∃ ρ. (Int × String × ρ)
  Supertype:
    ∃ ρ. (Int × ρ)
  Result:
    OK
tests/tuples02.types:4:1-4:49:
  Subtype:
    ∃ ρ. (Int × String × Float × ρ)
  Supertype:
    ∃ a. ∃ ρ. (Int × a × ρ)
  Result:
    OK